bin/ledger2beancount-txns

```diff
@@ -331,11 +333,20 @@ sub map_commodity($) {
     my ($key) = @_;
 
     return $config->{commodity_map}{$key} if exists $config->{commodity_map}{$key};
-    $key =~ s/"//g;
+    $key =~ s/^"//g;
+    $key =~ s/"$//g;
     # Check again after removing the quote
     return $config->{commodity_map}{$key} if exists $config->{commodity_map}{$key};
 
-    return uc $key;
+    # Generate a valid beancount name
+    # Dash (-) is not valid in ledger (even with quoted commodity) but valid
+    # in beancount
+    $key =~ s/[!"^& ]/-/g;
+
+    return $config->{commodity_map}{$key} if exists $config->{commodity_map}{$key};
+
+    # Maxium limit for beancount commodities: 24 characters
+    return substr (uc $key, 0, 24);
```

Comment:
we should probably emit a warning on STDERR when this happens, as it might cause collisions and that would be bad

Reply:
It would be easy to emit a warning EACH time but this sucks. It's harder to emit it once. But in the future, I'd like to embed some "notes" in the converted file that users have to look at it. And this would be a good note. Can you open an issue?

Reply:
done: #57